geode-core/src/main/java/org/apache/geode/internal/cache/Oplog.java

```diff
@@ -2225,6 +2225,12 @@ long recoverCrf(OplogEntryIdSet deletedIds, boolean recoverValues, boolean recov
         if (getParent().isOfflineCompacting()) {
           getParent().incLiveEntryCount(getRecoveryMap().size());
         }
+        long tc = totalCount.get();
+        long tlc = totalLiveCount.get();
+        if (getParent().isValidating() && tlc >= 0
+            && tc > tlc) {
```

Comment:
Can there be a case where tc and tlc both are same number (including 0); in that case dead-record-counts are 0; is this valid?

Reply:
Yes. It is valid. It is very common for tc(totalCount) and tlc(totalLiveCount) to be the same value, including 0, which indicates an empty Oplog. When tc and tlc are the same, offline disk store validation will not show any message about  the number of records that would be removed if you compacted the store, since there is no deleted or skipped entry in this case.